[Tests] : Add unit tests for logger.go in pkg/logger

Dhiren-Mhatre opened the pull request:
# [Feat] Add Unit Tests for pkg/logger Package

This PR adds comprehensive unit tests for the logger package to improve test coverage in the pkg/ directory. The changes include tests for both generic and dedicated logger implementations, ensuring all logging levels and methods are properly tested. This contribution is part of the larger effort to enhance test coverage across the codebase (#3835).

## How to use

Reviewers can validate this PR by:
1. Running the unit tests:
```bash
go test ./pkg/logger -v -coverprofile=coverage.out
```

2. Examining the coverage report:
```bash
go tool cover -html=coverage.out
```

3. Verifying that all logging methods and levels are properly tested:
- Standard logging levels (Panic to Trace)
- Both formatted and unformatted logging methods
- Level getter/setter functionality

## Testing done

Added comprehensive test suite with the following results:

```bash
=== RUN   TestNewFromGenericLogger
=== RUN   TestNewFromGenericLogger/Panic
=== RUN   TestNewFromGenericLogger/Fatal
[... other test outputs ...]
--- PASS: TestNewFromGenericLogger (0.00s)
=== RUN   TestNewFromDedicatedLogger
--- PASS: TestNewFromDedicatedLogger (0.00s)
=== RUN   TestDefaultLogger
--- PASS: TestDefaultLogger (0.00s)
PASS
```

The tests achieve full coverage of the logger package, including all logging levels and methods, with proper error handling and level management.
 
![Screenshot from 2025-02-04 00-17-59](https://github.com/user-attachments/assets/1a4e664a-ba18-44cd-93a3-a82f9e0bc2b9)

yashpandey06 commented:
@Dhiren-Mhatre  Did you run lint via makefile ?

Dhiren-Mhatre commented:
@yashpandey06 I ran make lint and only encountered a Docker-related warning about InvalidDefaultArgInFrom in the linter Dockerfile, which is unrelated to the logger tests.

yashpandey06 commented:
Did make run completed successfully or exited in.between ? ...can you share the logs ss

Dhiren-Mhatre commented:
@yashpandey06  Sure.
![image](https://github.com/user-attachments/assets/5988b894-8f0d-4e32-96c4-32cd04ca68fd)

yashpandey06 commented:
I am also getting invalid image url so is it working fine ? ...is it actually linting ?

Dhiren-Mhatre commented:
> @Dhiren-Mhatre Any prerequisites you did before running make lint actually I am trying the same but getting error logs .
I didn’t do any prerequisites before running make lint. It ran successfully for me, but I did get a warning about an invalid base image name. Are you seeing the same issue, or is it failing completely?

 
[Screencast from 2025-02-04 01-44-50.webm](https://github.com/user-attachments/assets/618f1032-bbc7-40c1-88c1-9b09ce23a947)

Dhiren-Mhatre commented:
@mauriciovasquezbernal, I notice my PR caused some website building issues that required fixes in PR #3989.Could you help me understand what in my logger tests caused the website building issue? This would help me prevent similar issues in future PRs

mauriciovasquezbernal commented:
> @mauriciovasquezbernal, I notice my PR caused some website building issues that required fixes in PR #3989.Could you help me understand what in my logger tests caused the website building issue? This would help me prevent similar issues in future PRs

That was a typo on my side, I intended to link #3948, which has a very similar number to this one.

Dhiren-Mhatre commented:
> > @mauriciovasquezbernal, I notice my PR caused some website building issues that required fixes in PR #3989.Could you help me understand what in my logger tests caused the website building issue? This would help me prevent similar issues in future PRs
> 
> That was a typo on my side, I intended to link #3948, which has a very similar number to this one.

@mauriciovasquezbernal Thank you for the clarification.

Dhiren-Mhatre commented:
@eiffel-fl  PTAL

burak-ok commented on pkg/logger/logger_test.go:
```diff
@@ -0,0 +1,145 @@
+	logger := NewFromGenericLogger(mock)
+	require.NotNil(t, logger)
+
+	// Test all logging methods
```
Could you also check the message that is getting logged?

burak-ok commented on pkg/logger/logger_test.go:
```diff
@@ -0,0 +1,145 @@
+import (
+	"testing"
+
+	"github.com/sirupsen/logrus"
```
Since we import `logrus` everywhere as `log` could you change that?

Dhiren-Mhatre commented:
@burak-ok Thanks for the review! I have made these changes:
used require.Same() for exact logrus logger instance check
Removed redundant TestLevelGetterSetter
Added message content verification in tests
Changed logrus import to use log alias
Let me know if you'd like any other improvements!

burak-ok commented on pkg/logger/logger_test.go:
```diff
@@ -0,0 +1,197 @@
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
```
Use `assert` instead of `require` when the test still can run without problems. This way we get multiple assert errors in the log instead of 1 by 1 after running and fixing them
```suggestion
	"github.com/stretchr/testify/assert"
```

Dhiren-Mhatre replied:
I'll use assert instead of require here after, thanks for the review!

burak-ok reviewed (commented):
Looks good. After that comment I think this is good to merge

burak-ok commented:
Since the second commit fixes the first one and is not a logical separate commit, they should be squashed in this PR

burak-ok reviewed (approved):
Thanks for your contribution
The failed documentation check is because of a temporary dead link it seems
